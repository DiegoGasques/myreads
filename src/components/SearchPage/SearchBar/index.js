import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import "./styles.css";

function SearchBar({ handleSearch }) {
  return (
    <div className="search-books-bar">
      <Link to="/" className="close-search">
        Close
      </Link>
      <div className="search-books-input-wrapper">
        <input
          type="text"
          placeholder="Search by title or author"
          onKeyUp={e => handleSearch(e.target.value)}
        />
      </div>
    </div>
  );
}

SearchBar.propTypes = {
  handleSearch: PropTypes.func.isRequired
};

export default SearchBar;
