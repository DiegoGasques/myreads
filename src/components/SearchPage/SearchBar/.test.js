import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import SearchBar from "./index";

it("Should match SearchBar component snapshot", () => {
  const handleSearchSpy = jest.fn();
  const wrapper = shallow(<SearchBar handleSearch={handleSearchSpy} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("Should call handleSearch prop func on input keyUp event with right value", () => {
  const value = "travel";
  const handleSearchSpy = jest.fn();
  const wrapper = shallow(<SearchBar handleSearch={handleSearchSpy} />);
  wrapper.find("input").simulate("keyUp", {
    target: {
      value
    }
  });
  expect(handleSearchSpy).toHaveBeenCalledWith(value);
});
