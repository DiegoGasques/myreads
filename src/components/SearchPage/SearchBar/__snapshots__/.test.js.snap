// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match SearchBar component snapshot 1`] = `
<div
  className="search-books-bar"
>
  <Link
    className="close-search"
    replace={false}
    to="/"
  >
    Close
  </Link>
  <div
    className="search-books-input-wrapper"
  >
    <input
      onKeyUp={[Function]}
      placeholder="Search by title or author"
      type="text"
    />
  </div>
</div>
`;
